### ISSUE TITLE
Django 1.8.4 regression regarding related model save.

### ISSUE THREAD

**Issue by coady** [django-tastypie/django-tastypie#1325]
The fix for django ticket ["Move unsaved model instance assignment check to model.save()"](https://code.djangoproject.com/ticket/25160) changes the related model check (again) and has broken this logic from `Resource.full_hydrate`:

```
                            # NOTE: A bug fix in Django (ticket #18153) fixes incorrect behavior
                            # which Tastypie was relying on.  To fix this, we store value.obj to
                            # be saved later in save_related.
                            try:
                                setattr(bundle.obj, field_object.attribute, value.obj)
                            except (ValueError, ObjectDoesNotExist):
                                bundle.related_objects_to_save[field_object.attribute] = value.obj
```

The `ValueError` will no longer be raised by django if the pk is None;  it raises later in the save method.  This results in tastypie creating a new related object even if it already exists.

**Comment by georgedorn** [comment]
Has that patched been released in 1.8?  It's unclear from the trac ticket.

**Comment by acoady** [comment]
Yes, the [patch was backported](https://github.com/django/django/commit/e4b813c0e723e774c5c7386a29f10bc807a47e47) and [released in 1.8.4](https://docs.djangoproject.com/en/1.8/releases/1.8.4/).
